docs/Minimal_Build__Advanced_Usage_with_NNAPI.md

```diff
@@ -0,0 +1,121 @@
+- At the 'extended' level we additionally fuse the Conv and Relu nodes using the internal ONNX Runtime FusedConv operator.
+
+<img align="center" src="images/mnist_optimization.png" alt="Changes to nodes from basic and extended optimizations."/>
+
+During runtime, for each group of nodes that is assigned to it, the NNAPI EP will create an NNAPI model that replicates the processing done by the nodes in the group. It will create a function to handle executing this NNAPI model, including copying data between the CPU and NNAPI. The original nodes in the model are replaced with a single node containing this function.
+
+### Considerations when creating an NNAPI-aware ORT format model
+
+The ONNX model intended for using with NNAPI should be optimized when creating the ORT formatted model. *Step 1* and *Step 2* optimizations should be used. *Step 3* optimizations may be considerd based on the performance impacts as the benefits are model dependent.
+
+*Step 2* optimizations would identify and assign nodes for NNAPI execution and replace groups of nodes with functions when possible. This will contribute towards improving the performance by reducing data flow between CPU and NNAPI.
+
+The 'extended' optimizations from *Step 3* may benefit the ORT model as some of the CPU nodes can be replaced by the custom ONNX operators designed for improved execution on the CPU. 
+
+The 'layout' optimizations are not applicable for execution with NNAPI.
+
+Below is an example for the MNIST model comparing the ORT format model with 'basic' or 'extended' optimizations.
+
+As the 'basic' level optimizations result in a model that only uses ONNX operators, the NNAPI EP is able to handle the majority of the model in a single function as NNAPI can execute all the Conv, Relu and MaxPool nodes at once.
+
+The 'extended' level optimizations introduced the custom FusedConv nodes, resulting in two functions using NNAPI, each handling a single MaxPool node. 
+
+<img align="center" src="images/mnist_optimization_with_nnapi.png" alt="Changes to nodes by NNAPI depending on optimization level of input.">
+
+
+In this example, the 'extended' optimizations resulted in worse performance with NNAPI due to the additional data copies between CPU and NNAPI for the MaxPool calculations.
+
+__NOTE:__ If your model benefits signficantly from 'extended' optimizations, you may wish to create an ORT format model optimized to 'basic' level for use with NNAPI (see instructions [here](ONNX_Runtime_for_Mobile_Platforms.md#Using-NNAPI-with-ONNX-Runtime-Mobile)), and an ORT format model optimized to 'extended' level for use everywhere else.
+
+If your model benefits from 'extended' optimizations then it is possible to create an ORT format model that only applies extended optimizations to nodes assigned for CPU execution, i.e. not handled by NNAPI.
+
```

Comment:
These 2 are bit confusing (they both state "If your model benefits signficantly from [extended] optimizations", with difference actions) , maybe change the 2nd to something like 
"You may choose to create an ORT format model that only applies extended optimizations to nodes assigned for CPU execution, i.e. not handled by NNAPI."